R/utils.R

```diff
@@ -578,101 +562,39 @@ wait_for_nonmem.bbi_nonmem_model <- function(mod, time_limit = 200, interval = 5
-#' @param interval integer for number of seconds to wait between each check
-#'
-#' @importFrom purrr map_lgl
-#' @importFrom checkmate assert_list
-#'
 #'
 #' @describeIn wait_for_nonmem takes a `list` of `bbi_nonmem_model` objects.
 #' @export
 wait_for_nonmem.list <- function(mod, time_limit = 200, interval = 5) {
 
   assert_list(mod)
+  check_model_object_list(mod)
+  verbose_msg(glue("Waiting for {length(mod)} model(s) to finish..."))
 
-  expiration <- Sys.time() + time_limit
   Sys.sleep(1) # wait for lst file to be created
-
+  expiration <- Sys.time() + time_limit
+  n_interval <- 0
   while ((expiration - Sys.time()) > 0) {
     res <- map_lgl(mod, ~check_nonmem_finished(.x))
     if (all(res)) {
       break
+    }else{
+      n_interval = n_interval + 1
+      # print message every 10 intervals
+      if(n_interval %% 10 == 0){
+        verbose_msg(glue("Waiting for {length(res[!res])} model(s) to finish..."))
+      }
     }
     Sys.sleep(interval)
   }
-}
-
-
 
-#' R wrapper for `qstat`
-#'
-#' @details
-#'
-#' Note: This function was designed to be utilized on `Metworx`. Functionality is not guaranteed if running on another platform.
-#'
-#' The returned dataframe will look identical to what you see in `qstat`, minus the last column (`ja-task-ID`).
-#'
-#' if all jobs are in the queue (i.e. no running jobs), the `slots` column will appear as `NA` and a warning will be thrown.
-#'
-#'
-#' @returns A dataframe
-#'
-#' @importFrom tidyr separate
-#' @export
-fetch_model_runs <- function(){
-  command <- "qstat -xml | tr '\n' ' ' | sed 's#<job_list[^>]*>#new_line#g' | sed 's#<[^>]*>##g' | grep ' ' | column -t"
-  running_models <- system(command, intern = TRUE)
-
-  if(is_empty(running_models)) return(data.frame())
-
-  runs <- strsplit(running_models, "new_line")[[1]][-1] %>%
-    str_trim()
-  one_space <- gsub("\\s+"," ",runs,fixed = F)
-
-  runs_df <- as.data.frame(one_space) %>%
-    separate(col = "one_space",
-                    into = c("job-ID", "prior", "name", "user", "state",
-                             "submit/start at", "queue", "slots"),
-                    sep = " ")
-
-  return(runs_df)
-}
-
-#' Crash a specific job on the grid
-#'
-#' @description
-#' Wrapper for `qdel`
-#'
-#' @details
-#'
-#' It is advised to run [fetch_model_runs()] (or `qstat` in your terminal) before calling this function to avoid accidentally deleting the wrong job.
-#'
-#' @param run_name character vector. The `name(s)` of the job(s) you want to end.
-#'
-#' @examples
-#' \dontrun{
-#' crash_model_run("Run_1")
-#' crash_model_run(c("Run_1", "Run_4"))
-#' }
-#'
-#' @importFrom rlang abort
-#'
-#' @export
-crash_model_run <- function(run_name){
-
-  # warning message occurs if slots are NA (only jobs in the queue)
-  runs <- suppressWarnings(fetch_model_runs())
-
-  if(is_empty(runs)){
-    stop("No jobs running at this time")
+  if(expiration < Sys.time() & !map_lgl(mod, ~check_nonmem_finished(.x))){
```

Comment:
You're gonna want `&&` here. See explanation [here](https://stackoverflow.com/a/6559049/3743213) for details (essentially we only want this to return a single boolean, not a vector of them).

But also, this causes a warning whenever you pass more than one model:

```
Warning message:
In if (expiration < Sys.time() & !map_lgl(mod, ~check_nonmem_finished(.x))) { :
  the condition has length > 1 and only the first element will be used
```

Reply:
Good call here. I've never observed a difference in results between the two (despite knowing the differences), but this is the first example I've seen where its absolutely warranted